src/Controller/Api/AbstractController.php

```diff
@@ -34,67 +34,54 @@
 
 class AbstractController extends \Symfony\Bundle\FrameworkBundle\Controller\AbstractController
 {
-    /**
-     * @var \JMS\Serializer\Serializer
-     */
-    protected $serializer;
+    protected SerializerInterface $serializer;
 
     public function __construct(SerializerInterface $serializer)
     {
         $this->serializer = $serializer;
     }
 
-    /**
-     * @param string $version
-     * @return \App\Entity\MajorVersion
-     */
     protected function findMajorVersion(string $version): MajorVersion
     {
         $this->checkMajorVersionFormat($version);
-        $repo = $this->getDoctrine()->getRepository(MajorVersion::class);
-        $entity = $repo->findOneBy(['version' => $version]);
-        if (null === $entity) {
+        $majorVersion = $this->getDoctrine()->getRepository(MajorVersion::class)->findOneBy(['version' => $version]);
+        if (!$majorVersion instanceof MajorVersion) {
             throw new NotFoundHttpException('No such version.');
         }
-        return $entity;
+        return $majorVersion;
     }
 
     /**
-     * @param \Symfony\Component\Validator\Validator\ValidatorInterface $validator
      * @param $object
      */
     protected function validateObject(ValidatorInterface $validator, $object): void
     {
-        $errors = $validator->validate($object);
+        $constraintViolationList = $validator->validate($object);
 
-        if (\count($errors) > 0) {
-            $errorsString = (string)$errors;
+        if (\count($constraintViolationList) > 0) {
+            $errorsString = (string)$constraintViolationList;
             throw new BadRequestHttpException($errorsString);
         }
     }
 
     protected function mapObjects($baseObject, array $data): void
```

Comment:
Also, `$data` is a poor variable name.  I'm kind of surprised Rector didn't complain about it, given the other things it was complaining about.  Can that be more descriptive?  It needs a docblock description, too.  (I know Symfony often doesn't like docblocks, but they should be used if something is not readily obvious from the method signature, and in this case this method's purpose and inputs are not at all obvious.)

Reply:
Agree, will check it...